tests/pytests/test_acl.py

```diff
@@ -0,0 +1,77 @@
+from common import *
+
+def test_acl_category(env):
+    """Test that the `search` category was added appropriately in module
+    load"""
+    res = env.cmd('ACL', 'CAT')
+    env.assertTrue('search' in res)
+
+@skip(redis_less_than="7.4.1")
+def test_acl_search_commands(env):
+    """Tests that the RediSearch commands are registered to the `search`
+    ACL category"""
+    res = env.cmd('ACL', 'CAT', 'search')
+    # TODO: Do we want to hide the internal commands (e.g., "_FT.*")?
+    commands = [
+        'FT.EXPLAINCLI', '_FT.ALIASDEL', 'FT.SPELLCHECK', 'FT.SYNUPDATE',
+        'FT.ALIASUPDATE', '_FT.AGGREGATE', '_FT.ALIASADD', 'FT._LIST',
+        '_FT.ALIASUPDATE', 'FT.ALIASADD', 'FT.SEARCH',
+        '_FT.CURSOR', 'FT.INFO', 'FT.SUGDEL', '_FT.INFO', '_FT.SUGDEL',
+        '_FT.ALTER', '_FT.DICTDEL', '_FT.SYNUPDATE', 'FT.DICTDUMP',
+        'FT.EXPLAIN', '_FT.SPELLCHECK', 'FT.AGGREGATE', 'FT.SUGLEN',
+        '_FT.SUGLEN', 'FT.PROFILE', 'FT.ALTER', 'FT.SUGGET', '_FT.CREATE',
+        'FT.DICTDEL', 'FT.CURSOR', 'FT.ALIASDEL', 'FT.SUGADD', '_FT.DICTADD',
+        'FT.SYNDUMP', 'FT.CREATE', '_FT.PROFILE', '_FT.SEARCH', 'FT.DICTADD',
+        'FT.SYNFORCEUPDATE', 'FT._ALIASDELIFX', 'FT._CREATEIFNX',
+        '_FT.DEBUG', '_FT.CONFIG', '_FT.TAGVALS', '_FT.SUGGET',
+        'search.CLUSTERREFRESH', 'FT._ALIASADDIFNX', '_FT._ALTERIFNX',
+        '_FT.SUGADD', 'FT._ALTERIFNX', '_FT._ALIASDELIFX', 'search.CLUSTERSET',
+        'search.CLUSTERINFO', '_FT._ALIASADDIFNX', 'FT.CONFIG'
+    ]
+    # Use a set since the order of the response is not consistent.
+    commands = set(commands)
+    env.assertEqual(set(res), commands)
+
+    # Check that one of our commands is listed in a non-search category
+    res = env.cmd('ACL', 'CAT', 'read')
+    env.assertTrue('FT.SEARCH' in res)
+
+def test_acl_non_default_user(env):
+    """Tests that a user with a non-default ACL can't access the search
+    category"""
+    # Create a user with no command permissions
+    env.expect('ACL', 'SETUSER', 'test', 'on', '>123', '~*', '&*').ok()
```

Comment:
Consider changing "test" -> "test-user"
Also if the `>123`, `-*` and `&*` have meaning then move them to be variables so they will have some context and it will be easier to maintain.

Reply:
`123` is the password, and the other indicators indicate the key, pubsub and command permissions for the new ACL user. Since these are dynamic and change from user to user (i.e., not reused mostly, and specifically in this test), I think they should stay as they are.

Reply:
If you don't want to store the values in variables them please add a comment explaining them.
Especially the pubsub and command permissions.

Reply:
There is already an explanation above. That's the only one relevant for the test.

Reply:
I can add more if it's very important to you, but I think this is enough.

Reply:
Added